keyboards/doro67/rgb/rgb.c

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2019 MechMerlin
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rgb.h"
+
+// Optional override functions below.
+// You can leave any or all of these undefined.
+// These are only required if you want to perform custom actions.
+
+
+
+void matrix_init_kb(void) {
+  // put your keyboard start-up code here
+  // runs once when the firmware starts up
+  setPinOutput(E6);
+  matrix_init_user();
+}
+
+void matrix_scan_kb(void) {
+  // put your looping keyboard code here
+  // runs every cycle (a lot)
+
+  matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+  // put your per-action keyboard code here
+  // runs for every action, just before processing by the firmware
+
+  return process_record_user(keycode, record);
+}
+
+void led_set_kb(uint8_t usb_led) {
+	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+  if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+    writePinLow(E6);
+  } else {
+    writePinHigh(E6);
+  }
+	led_set_user(usb_led);
+}
+
+#define NUMBER_OF_COLS 15
+#define NUMBER_OF_ROWS 5
+#define colCalc(col)  ((224/(NUMBER_OF_COLS - 1) * col))
+#define rowCalc(row)  ((64/(NUMBER_OF_ROWS - 1) * row))
+
+const rgb_led g_rgb_leds[DRIVER_LED_TOTAL] = {
+    {{0|(0<<4)},   {colCalc(0), rowCalc(0)},  0},
+    {{0|(1<<4)},   {colCalc(1), rowCalc(0)},  0},
+    {{0|(2<<4)},   {colCalc(2), rowCalc(0)},  0},
+    {{0|(3<<4)},   {colCalc(3), rowCalc(0)},  0},
+    {{0|(4<<4)},   {colCalc(4), rowCalc(0)},  0},
+    {{0|(5<<4)},   {colCalc(5), rowCalc(0)},  0},
+    {{0|(6<<4)},   {colCalc(6), rowCalc(0)},  0},
+    {{0|(7<<4)},   {colCalc(7), rowCalc(0)},  0},
+    {{0|(8<<4)},   {colCalc(8), rowCalc(0)},  0},
+    {{0|(9<<4)},   {colCalc(9), rowCalc(0)},  0},
+    {{0|(10<<4)},  {colCalc(10), rowCalc(0)}, 0},
+    {{0|(11<<4)},  {colCalc(11), rowCalc(0)}, 0},
+    {{0|(12<<4)},  {colCalc(12), rowCalc(0)}, 0},
+    {{0|(13<<4)},  {colCalc(13), rowCalc(0)}, 1},
+    {{0|(14<<4)},  {colCalc(14), rowCalc(0)}, 1},
+
+    {{1|(0<<4)},   {colCalc(0), rowCalc(1)},  1},
```

Comment:
Honestly, I don't think this is the best way.  It may be better to actually do the math and input the values. 

Specifically, because the rows are staggered, you will want to offset these slightly from the row above. 

Though, these are "backwards".... you'll see how the X position is slightly offset on each row?  
https://github.com/qmk/qmk_firmware/blob/master/keyboards/dztech/dz60rgb/dz60rgb.c#L67-L128

That's what you want to do, ideally.